src/cascadia/TerminalApp/TabManagement.cpp

```diff
@@ -824,6 +827,37 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Close all other panes except the pane that is currently focused.
+    // Arguments:
+    // - weakTab: weak reference to the tab that the pane belongs to.
+    // - paneIds: collection of the IDs of the panes that are marked for removal.
+    winrt::fire_and_forget TerminalPage::_CloseUnfocusedPanes(weak_ref<TerminalTab> weakTab, std::deque<uint32_t> paneIds)
+    {
+        if (auto strongTab{ weakTab.get() })
+        {
+            // Close all unfocused panes one by one
+            while (!paneIds.empty())
+            {
+                const auto id = paneIds.front();
+                paneIds.pop_front();
+
+                if (const auto pane{ strongTab->GetRootPane()->FindPane(id) })
+                {
+                    pane->ClosedByParent([ids{ std::move(paneIds) }, weakThis{ get_weak() }, weakTab]() {
+                        if (auto strongThis{ weakThis.get() })
+                        {
+                            strongThis->_CloseUnfocusedPanes(weakTab, std::move(ids));
+                        }
+                    });
+
+                    // Close the pane which will eventually trigger the closed by parent event
+                    co_await _HandleClosePaneRequested(pane);

```

Comment:
Why do we have to wait for the closed by parent event? Can't we close them all at once?

Reply:
Waiting for this call is indeed not necessary here (changed in f405ce0).

I am not entirely sure what you mean with the statement below in the given context. From my understanding, the pane is removed from the tree structure after the animation finishes (if enabled). This operation changes the structure of the tree  where, in some cases, values from the child pane are being moved into the parent pane. Removing the panes all at once would, in some cases, lead to panes not being closed properly.

> Can't we close them all at once?